cpp/ql/src/semmle/code/cpp/models/implementations/Memset.qll

```diff
@@ -0,0 +1,28 @@
+import semmle.code.cpp.Function
+import semmle.code.cpp.models.interfaces.ArrayFunction
+import semmle.code.cpp.models.interfaces.DataFlow
+import semmle.code.cpp.models.interfaces.Taint
+
+/**
+ * The standard function `memset` and its assorted variants
+ */
+class MemsetFunction extends ArrayFunction, DataFlowFunction, TaintFunction {
```

Comment:
You can remove `TaintFunction` from the list of superclasses and remove `Taint` from the list of imports.

Reply:
Or add taint flow from the character and number parameters to the buffer???  An attacker with control of them has some (very limited) control of the data that ends up in the buffer and could plausibly use this for harm.

Reply:
I argued against that in https://github.com/Semmle/ql/pull/1933#discussion_r327962404.

@zlaski-semmle and I just had a discussion about what constitutes taint. Here is the definition I wrote a long time ago: https://github.com/Semmle/ql/blob/e1594a4b0b9ff0cd3fe70e873ad38a98c0a6b41d/cpp/ql/src/semmle/code/cpp/dataflow/TaintTracking.qll#L5-L8. For other languages, like Java, it's completely different, and they haven't attempted to write down a definition. I think the only "definition" that fits all languages is: _taint should flow where it's beneficial for some taint-tracking queries and harmful for none_.